src/ciflow-push-trigger.ts

```diff
@@ -0,0 +1,169 @@
+import {Context, Probot} from 'probot';
+
+function isCIFlowLabel(label: string): boolean {
+  return label.startsWith('ciflow/');
+}
+
+function labelToTag(label: string, prNum: number): string {
+  return `${label}/${prNum}`;
+}
+
+function getAllPRTags(context: Context<'pull_request'>) {
+  const prNum = context.payload.pull_request.number;
+  const labels = context.payload.pull_request.labels
+    .map(label => label.name)
+    .filter(isCIFlowLabel);
+
+  context.log.info(labels, 'Found labels on PR');
+  return labels.map(label => labelToTag(label, prNum));
+}
+
+/**
+ * Make sure `tag` points to `head_sha`, deleting old tags as necessary.
+ * @param tag  looks like "ciflow/trunk/12345", where 12345 is the PR number.
+ * @param headSha
+ */
+async function syncTag(
+  context: Context<'pull_request'>,
+  tag: string,
+  headSha: string
+) {
+  context.log.info(`Synchronizing tag ${tag} to head sha ${headSha}`);
+  const matchingTags = await context.octokit.git.listMatchingRefs(
+    context.repo({ref: `tags/${tag}`})
+  );
+  if (matchingTags.data.length > 0) {
+    context.log.info(matchingTags.data, 'Found matching tags');
+  } else {
+    context.log.info(`No matching tags`);
+  }
+  for (const match of matchingTags.data) {
+    if (match.object.sha === headSha) {
+      context.log.info(`Tag ${tag} already points to ${headSha}`);
+      return;
+    }
+
+    context.log.info(
+      `deleting out of date tag ${tag} on sha ${match.object.sha}`
+    );
+    await context.octokit.git.deleteRef(context.repo({ref: `tags/${tag}`}));
+  }
+
+  context.log.info(`Creating tag ${tag} on head sha ${headSha}`);
+  await context.octokit.git.createRef(
+    context.repo({ref: `refs/tags/${tag}`, sha: headSha})
+  );
+}
+
+/**
+ * Remove a tag from the repo if necessary.
+ * @param tag  looks like "ciflow/trunk/12345", where 12345 is the PR number.
+ */
+async function rmTag(context: Context, tag: string) {
+  context.log.info(`Cleaning up tag ${tag}`);
+  const matchingTags = await context.octokit.git.listMatchingRefs(
+    context.repo({ref: `tags/${tag}`})
+  );
+  for (const match of matchingTags.data) {
+    if (match.ref === `refs/tags/${tag}`) {
+      context.log.info(`Deleting tag ${tag} on sha ${match.object.sha}`);
+      await context.octokit.git.deleteRef(context.repo({ref: `tags/${tag}`}));
+      return;
+    }
+  }
+  context.log.info(`No matching tags for ${tag}`);
```

Comment:
Do we need all these context.log.info and context.log.debug messages in production code ?

Reply:
these are for logging on the serverless function, so that if something goes wrong we can understand what happened.